### PR TITLE
Fix a loop pointer bug in EnsureBackendPoolDeleted

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix a loop pointer bug in EnsureBackendPoolDeleted
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix nic CreateOrUpdate bug when there are more than one nic.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lzhecheng** [comment]
Without this fix, there'll be ccm log like:
```
I0217 16:34:57.669834       1 azure_standard.go:1170] EnsureBackendPoolDeleted begins to CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) with backendPoolIDs ["/subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/loadBalancers/capz-1lmk86-internal/backendAddressPools/capz-1lmk86"]
I0217 16:34:57.670237       1 azure_standard.go:1170] EnsureBackendPoolDeleted begins to CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) with backendPoolIDs ["/subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/loadBalancers/capz-1lmk86-internal/backendAddressPools/capz-1lmk86"]
I0217 16:35:02.380778       1 route_controller.go:299] set node capz-1lmk86-md-0-hv6xj with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:02.380790       1 route_controller.go:299] set node capz-1lmk86-control-plane-mtgjt with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:02.380804       1 route_controller.go:299] set node capz-1lmk86-md-0-jskfk with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:06.334583       1 reflector.go:281] k8s.io/client-go/informers/factory.go:150: forcing resync
I0217 16:35:06.355673       1 reflector.go:281] k8s.io/client-go/informers/factory.go:150: forcing resync
I0217 16:35:06.558611       1 httplog.go:132] "HTTP" verb="GET" URI="/healthz" latency="87.801µs" userAgent="kube-probe/1.27+" audit-ID="" srcIP="10.0.0.4:59628" resp=200
E0217 16:35:08.368708       1 azure_standard.go:1173] EnsureBackendPoolDeleted CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) failed with error Retriable: false, RetryAfter: 0s, HTTPStatusCode: 412, RawError: {
  "error": {
    "code": "PreconditionFailed",
    "message": "Precondition failed.",
    "details": [
      {
        "code": "PreconditionFailedEtagMismatch",
        "message": "Etag provided in if-match header W/\"3da7bfdc-ed3f-47f4-a5dd-593e7bf4bcc2\" does not match etag W/\"c2106c0a-80d7-4459-814d-1e3832912d27\" of resource /subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/networkInterfaces/capz-1lmk86-md-0-jskfk-nic in NRP data store."
      }
    ]
  }
}I0217 16:34:57.669834       1 azure_standard.go:1170] EnsureBackendPoolDeleted begins to CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) with backendPoolIDs ["/subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/loadBalancers/capz-1lmk86-internal/backendAddressPools/capz-1lmk86"]
I0217 16:34:57.670237       1 azure_standard.go:1170] EnsureBackendPoolDeleted begins to CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) with backendPoolIDs ["/subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/loadBalancers/capz-1lmk86-internal/backendAddressPools/capz-1lmk86"]
I0217 16:35:02.380778       1 route_controller.go:299] set node capz-1lmk86-md-0-hv6xj with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:02.380790       1 route_controller.go:299] set node capz-1lmk86-control-plane-mtgjt with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:02.380804       1 route_controller.go:299] set node capz-1lmk86-md-0-jskfk with NodeNetworkUnavailable=false was canceled because it is already set
I0217 16:35:06.334583       1 reflector.go:281] k8s.io/client-go/informers/factory.go:150: forcing resync
I0217 16:35:06.355673       1 reflector.go:281] k8s.io/client-go/informers/factory.go:150: forcing resync
I0217 16:35:06.558611       1 httplog.go:132] "HTTP" verb="GET" URI="/healthz" latency="87.801µs" userAgent="kube-probe/1.27+" audit-ID="" srcIP="10.0.0.4:59628" resp=200
E0217 16:35:08.368708       1 azure_standard.go:1173] EnsureBackendPoolDeleted CreateOrUpdate for NIC(capz-1lmk86, capz-1lmk86-md-0-jskfk-nic) failed with error Retriable: false, RetryAfter: 0s, HTTPStatusCode: 412, RawError: {
  "error": {
    "code": "PreconditionFailed",
    "message": "Precondition failed.",
    "details": [
      {
        "code": "PreconditionFailedEtagMismatch",
        "message": "Etag provided in if-match header W/\"3da7bfdc-ed3f-47f4-a5dd-593e7bf4bcc2\" does not match etag W/\"c2106c0a-80d7-4459-814d-1e3832912d27\" of resource /subscriptions/===REDACTED===/resourceGroups/capz-1lmk86/providers/Microsoft.Network/networkInterfaces/capz-1lmk86-md-0-jskfk-nic in NRP data store."
      }
    ]
  }
}
```
https://storage.googleapis.com/kubernetes-jenkins/logs/cloud-provider-azure-master-capz/1626612592988393472/artifacts/clusters/capz-1lmk86/kube-system/cloud-controller-manager-7b484f44c5-jndnt/cloud-controller-manager.log

File: pkg/provider/azure_standard.go [hunk 1]
```diff
@@ -1142,7 +1142,8 @@ func (as *availabilitySet) EnsureBackendPoolDeleted(service *v1.Service, backend
 			ipconfigPrefixToNicMap[ipConfigIDPrefix] = nic
 		}
 	}
-	for _, nic := range ipconfigPrefixToNicMap {
+	for k := range ipconfigPrefixToNicMap {
```
Comment (feiskyer) [inline pkg/provider/azure_standard.go:1145]: Good catch. Is there any golint rules could be enabled to catch such issues?

Comment (lzhecheng) [reply pkg/provider/azure_standard.go:1145]: I think we should have it: golangci-lint but somehow it isn't working here. I'll investigate how to solve it.

Comment (andyzhangx) [reply pkg/provider/azure_standard.go:1145]: that's a classic pointer bug in go range statement, it's interesting how to catch it by some scanning tool.

**Pull request merged**